turbinia/lib/google_cloud.py

```diff
@@ -14,31 +13,61 @@
 # limitations under the License.
 """Google Cloud resources library."""
 
-from __future__ import unicode_literals
 import logging
+import traceback
 
-from google.cloud import error_reporting
-from google.cloud import exceptions
+import google.auth
+import googleapiclient.discovery
 
-from turbinia import TurbiniaException
+from turbinia import __version__
 
 logger = logging.getLogger('turbinia')
 
 
-def setup_stackdriver_traceback(project_id):
-  """Set up Google Cloud Error Reporting
+class GCPErrorReporting:
+  """This class is used to report errors to Google Cloud."""
 
-  This method will enable Google Cloud Error Reporting.
-  All exceptions that occur within a Turbinia Task will be logged.
+  def __init__(self):
+    self._credentials, self._project = self._create_credentials()
+    self.logging_client = googleapiclient.discovery.build(
+        'clouderrorreporting',
+        'v1beta1',
+        credentials=self._credentials,
+    )
 
-  Attributes:
-    project_id: The name of the Google Cloud project.
-  Raises:
-    TurbiniaException: When an error occurs enabling GCP Error Reporting.
-  """
-  try:
-    client = error_reporting.Client(project=project_id)
-  except exceptions.GoogleCloudError as exception:
-    msg = f'Error enabling GCP Error Reporting: {str(exception):s}'
-    raise TurbiniaException(msg)
-  return client
+  @staticmethod
+  def _create_credentials() -> google.auth.credentials.Credentials:
+    return google.auth.default()
+
+  def report(self, message, caller=None) -> None:
+    if not caller:
+      stack = traceback.extract_stack()
+      caller = stack[-2]
+    file_path = caller[0]
+    line_number = caller[1]
+    function_name = caller[2]
+    report_location = {
+        'filePath': file_path,
+        'lineNumber': line_number,
+        'functionName': function_name
+    }
+    # logger.warning(message, caller)
+    try:
+      self._send_error_report(message, report_location=report_location)
+    except Exception:
```

Comment:
Is there potentially a set of narrower exceptions we could catch rather than `Exception` here?

Reply:
Sort of. The API itself doesn't explicitly say what type of exceptions it can throw. Looked at the Python API client and it seems HttpError is the one that makes the most sense, but there are several from what I saw.